docs/tutorials/on-device-training/ios-app.md

```diff
@@ -0,0 +1,991 @@
+        )
+    }
+    
+    private func getORTValue(labels: [Int64]) throws -> ORTValue {
+        let tensorData = NSMutableData(bytes: labels, length: labels.count * MemoryLayout<Int64>.stride)
+        let inputShape: [NSNumber] = [labels.count as NSNumber]
+        
+        return try ORTValue (
+            tensorData: tensorData, elementType: ORTTensorElementDataType.int64, shape: inputShape
+        )
+    }
+    ```
+
+    c. We are ready to write the `trainStep` function, which takes a batch of input data and labels and performs one training step on the given batch.
+
+    ```swift
+    func trainStep(inputData: [Data], labels: [Int64]) throws  {
+        let inputs = [try getORTValue(dataList: inputData), try getORTValue(labels: labels)]
+        try trainingSession.trainStep(withInputValues: inputs)
+        
+        // update the model params
+        try trainingSession.optimizerStep()
+        
+        // reset the gradients
+        try trainingSession.lazyResetGrad()
+    }
+    ```
+
+    d. Finally, we have everything we need to write the training loop. Here, `kNumOtherRecordings` represent how many recordings we have in the `recordings` directory that we created earlier. `kNumEpochs` represents how many epochs we want to train the model on the given data. `kUserIndex` and `kOtherIndex` represent the labels for the user and other recordings respectively.
+
+    We also have a `progressCallback` that will be called after each training step. We will use this callback to update the progress bar in the UI.
+
+    ```swift
+    private let kNumOtherRecordings: Int = 20
+    private let kNumEpochs: Int = 3
+    
+    let kUserIndex: Int64 = 1
+    let kOtherIndex: Int64 = 0
+
+    func train(_ trainingData: [Data], progressCallback: @escaping (Double) -> Void) throws {
+        let numRecordings = trainingData.count
+        var otherRecordings = Array(0..<kNumOtherRecordings)
+        for e in 0..<kNumEpochs {
+            print("Epoch: \(e)")
+            otherRecordings.shuffle()
+            let otherData = otherRecordings.prefix(numRecordings)
+            
+            for i in 0..<numRecordings {
+                let (buffer, wavFileData) = try getDataFromWavFile(fileName: "other_\(otherData[i])")
+                try trainStep(inputData: [trainingData[i], wavFileData], labels: [kUserIndex, kOtherIndex])
+                print("finished training on recording \(i)")
+                
+                let progress = Double((e * numRecordings) + i + 1) / Double(kNumEpochs * numRecordings)
+                progressCallback(progress)
+            }
+        }
+        
+    }
+    ```
+
+3.  #### Exporting the trained model
+
+    We can use the `exportModelForInference` method of the `ORTTrainingSession` class to export the trained model. The method takes in the path where the model should be exported and the output names of the model.
```

Comment:
Is the name of the class `ORTTrainingSession` or `Trainer`?

Reply:
I meant that we can use `exportModelForInference` from the ort training API to export the model. 

https://onnxruntime.ai/docs/api/objectivec/Classes/ORTTrainingSession.html#/c:objc(cs)ORTTrainingSession(im)exportModelForInferenceWithOutputPath:graphOutputNames:error: